ocs_ci/ocs/ui/acm_ui.py

```diff
@@ -103,3 +130,487 @@ def navigate_credentials_page(self):
         """
         log.info("Navigate into Governance Page")
         self.do_click(locator=self.acm_page_nav["Credentials"])
+
+
+class ACMOCPClusterDeployment(AcmPageNavigator):
+    """
+    Everything related to cluster creation through ACM goes here
+
+    """
+
+    def __init__(self, driver, platform, cluster_conf):
+        super().__init__(driver)
+        self.platform = platform
+        self.cluster_conf = cluster_conf
+        self.cluster_name = self.cluster_conf.ENV_DATA["cluster_name"]
+        self.cluster_path = self.cluster_conf.ENV_DATA["cluster_path"]
+        self.deploy_sync_mode = config.MULTICLUSTER.get("deploy_sync_mode", "async")
+        self.deployment_status = None
+        self.cluster_deploy_timeout = self.cluster_conf.ENV_DATA.get(
+            "cluster_deploy_timeout", ACM_CLUSTER_DEPLOY_TIMEOUT
+        )
+        self.deployment_failed_reason = None
+        self.deployment_start_time = 0
+
+    def create_cluster_prereq(self):
+        raise NotImplementedError("Child class has to implement this method")
+
+    def navigate_create_clusters_page(self):
+        # Navigate to Clusters page which has 'Create Cluster'/
+        # 'Import Cluster' buttons
+        # Here we click on "Create Cluster" and we will be in create cluster page
+        while True:
+            self.navigate_clusters_page()
+            log.info("Clicking on 'CreateCluster'")
+            if not self.check_element_presence(
+                (By.XPATH, self.acm_page_nav["cc_create_cluster"][0]), timeout=200
+            ):
+                raise ACMClusterDeployException("Create cluster button not found")
+            log.info("Found create cluster")
+            if not self.check_element_presence(
+                (By.XPATH, self.acm_page_nav["cc_create_cluster_index_xpath"][0]),
+                timeout=300,
+            ):
+                raise ACMClusterDeployException("Create cluster button not found")
+            log.info("Found create cluster by index path")
+            self.do_click(locator=self.acm_page_nav["cc_create_cluster"], timeout=100)
+            time.sleep(20)
+            if self.driver.current_url.endswith("create-cluster"):
+                break
+
+    def click_next_button(self):
+        self.do_click(self.acm_page_nav["cc_next_page_button"])
+
+    def send_keys_multiple(self, key_val):
+        """
+        In a page if we want to fill multiple text boxes we can use
+        this function which iteratively fills in values from the dictionary parameter
+
+        key_val (dict): keys corresponds to the xpath of text box, value corresponds
+            to the value to be filled in
+
+        """
+        for xpath, value in key_val.items():
+            self.do_send_keys(locator=xpath, text=value)
+
+    def click_platform_and_credentials(self):
```

Comment:
I definitely need this function , because if I want to retry due to selenium slow response or sometimes totally misbehaving then functions makes my life easier.